manta-pay/src/test/compat.rs

```diff
@@ -0,0 +1,307 @@
+    let proving_context = MultiProvingContext {
+        mint: ProvingContext::decode(IoReader(File::open(mint_path)?))
+            .expect("Unable to decode MINT proving context."),
+        private_transfer: ProvingContext::decode(IoReader(File::open(private_transfer_path)?))
+            .expect("Unable to decode PRIVATE_TRANSFER proving context."),
+        reclaim: ProvingContext::decode(IoReader(File::open(reclaim_path)?))
+            .expect("Unable to decode RECLAIM proving context."),
+    };
+    let verifying_context = MultiVerifyingContext {
+        mint: VerifyingContext::decode(
+            manta_sdk::pay::testnet::verifying::Mint::get().expect("Checksum did not match."),
+        )
+        .expect("Unable to decode MINT verifying context."),
+        private_transfer: VerifyingContext::decode(
+            manta_sdk::pay::testnet::verifying::PrivateTransfer::get()
+                .expect("Checksum did not match."),
+        )
+        .expect("Unable to decode PRIVATE_TRANSFER verifying context."),
+        reclaim: VerifyingContext::decode(
+            manta_sdk::pay::testnet::verifying::Reclaim::get().expect("Checksum did not match."),
+        )
+        .expect("Unable to decode RECLAIM verifying context."),
+    };
+    let parameters = Parameters {
+        note_encryption_scheme: NoteEncryptionScheme::decode(
+            manta_sdk::pay::testnet::parameters::NoteEncryptionScheme::get()
+                .expect("Checksum did not match."),
+        )
+        .expect("Unable to decode NOTE_ENCRYPTION_SCHEME parameters."),
+        utxo_commitment: UtxoCommitmentScheme::decode(
+            manta_sdk::pay::testnet::parameters::UtxoCommitmentScheme::get()
+                .expect("Checksum did not match."),
+        )
+        .expect("Unable to decode UTXO_COMMITMENT_SCHEME parameters."),
+        void_number_commitment: VoidNumberCommitmentScheme::decode(
+            manta_sdk::pay::testnet::parameters::VoidNumberCommitmentScheme::get()
+                .expect("Checksum did not match."),
+        )
+        .expect("Unable to decode VOID_NUMBER_COMMITMENT_SCHEME parameters."),
+    };
+    println!("Loading parameters Done.");
+    Ok((
+        proving_context,
+        verifying_context,
+        parameters,
+        UtxoAccumulatorModel::decode(
+            manta_sdk::pay::testnet::parameters::UtxoAccumulatorModel::get()
+                .expect("Checksum did not match."),
+        )
+        .expect("Unable to decode UTXO_ACCUMULATOR_MODEL."),
+    ))
+}
+
+/// Asserts that `post` represents a valid `Transfer` verifying against `verifying_context`.
+#[inline]
+fn assert_valid_proof(verifying_context: &VerifyingContext, post: &config::TransferPost) {
+    assert!(
+        ProofSystem::verify(
+            verifying_context,
+            &post.generate_proof_input(),
+            &post.validity_proof,
+        )
+        .expect("Unable to verify proof."),
+        "Invalid proof: {:?}.",
+        post
+    );
+}
```

Comment:
The same function is also used in [manta-benchmark](https://github.com/Manta-Network/manta-rs/blob/main/manta-benchmark/src/payment.rs#L38-L51). Should we keep only one?